### DISCUSSION TITLE
docker-compose部署milvus:v2.3.11-gpu报错，nvidia-container-cli: initialization error: nvml error: driver/library version mismatch: unknown

### DISCUSSION THREAD

**Discussion by chenkang404** [milvus-io/milvus#31194] [Q&A and General discussion]
我的环境为：ubuntu v20.04,docker v24.0.4, docker-compose v2.24.6
采用的docker-compose up -d 的方式进行的milvus部署
报错为：
![image](https://github.com/milvus-io/milvus/assets/38779739/15617d96-e7b3-40c0-85d9-9e1bdda51c03)
用非gpu版本部署是可以的，但不能请求gpu
我的docker-compose.yml为：
version: '3.5'

services:
  etcd:
    container_name: milvus-etcd
    image: quay.io/coreos/etcd:v3.5.5
    environment:
      - ETCD_AUTO_COMPACTION_MODE=revision
      - ETCD_AUTO_COMPACTION_RETENTION=1000
      - ETCD_QUOTA_BACKEND_BYTES=4294967296
      - ETCD_SNAPSHOT_COUNT=50000
    volumes:
      - ${DOCKER_VOLUME_DIRECTORY:-.}/volumes/etcd:/etcd
    command: etcd -advertise-client-urls=http://127.0.0.1:2379 -listen-client-urls http://0.0.0.0:2379 --data-dir /etcd
    healthcheck:
      test: ["CMD", "etcdctl", "endpoint", "health"]
      interval: 30s
      timeout: 20s
      retries: 3

  minio:
    container_name: milvus-minio
    image: minio/minio:RELEASE.2023-03-20T20-16-18Z
    environment:
      MINIO_ACCESS_KEY: minioadmin
      MINIO_SECRET_KEY: minioadmin
    ports:
      - "9001:9001"
      - "9000:9000"
    volumes:
      - ${DOCKER_VOLUME_DIRECTORY:-.}/volumes/minio:/minio_data
    command: minio server /minio_data --console-address ":9001"
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost:9000/minio/health/live"]
      interval: 30s
      timeout: 20s
      retries: 3

  standalone:
    container_name: milvus-standalone
    image: milvusdb/milvus:v2.3.11-gpu
    command: ["milvus", "run", "standalone"]
    security_opt:
    - seccomp:unconfined
    environment:
      ETCD_ENDPOINTS: etcd:2379
      MINIO_ADDRESS: minio:9000
    volumes:
      - ${DOCKER_VOLUME_DIRECTORY:-.}/volumes/milvus:/var/lib/milvus
    ports:
        - "19530:19530"
        - "9091:9091"
    deploy:
      resources:
        reservations:
          devices:
            - driver: nvidia
              capabilities: ["gpu"]
              device_ids: ["2"]
    depends_on:
      - "etcd"
      - "minio"

networks:
  default:
    name: milvus

**Comment by chenkang404** [comment]
nvidia驱动版本，nvidia/cuda:11.6.2-base-ubuntu20.04,
但用命令nvidia-smi时，展示的gpu信息如下，
![Uploading image.png…]()

**Comment by presburger** [comment]
@chenkang404  Hi, please try to install nvidia container toolkit , ref https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/install-guide.html

**Reply by chenkang404** [reply]
ok，我最后排查原因应该是nvidia版本问题，但当前的nvidia版本还不能更换，只能先不用gpu了
